spyder/plugins/editor/widgets/editor.py

```diff
@@ -2329,6 +2329,14 @@ def __refresh_readonly(self, index):
             if not osp.isfile(finfo.filename):
                 # This is an 'untitledX.py' file (newly created)
                 read_only = False
+            try:
+                # Try to open the file to see if the permissions allow writing
+                # in Windows. For further info, see issue
+                # https://github.com/spyder-ide/spyder/issues/10657
+                fd = open(finfo.filename, os.O_RDWR)
+                fd.close()

```

Comment:
@steff456 here the `open`/`close` that we want to call is from `os` not the built-in one:
```suggestion
                fd = os.open(finfo.filename, os.O_RDWR)
                os.close(fd)
```

Reply:
This why the tests are failing with a message of `open` not accepting `int` type as second parameter  (`TypeError: open() argument 2 must be str, not int`)